File: src/Task2.java
public final class Task2 {
    private static int[] primesTo(int k) {
        int[] numbers = new int[k - 1];
        
        for (int i = 0; i < numbers.length; i++) {
            numbers[i] = i + 2;
        }
        
        for (int i = 0; i < numbers.length; i++) {
            int number = numbers[i];
            if (number != 0) {
                for (int j = i + number; j < numbers.length; j += number) {
                    numbers[j] = 0;
                }
            }
        }
        
        return numbers;
    }
    
    public static void main(String[] args) {
        for (int prime : primesTo(100000)) {
            if (prime != 0) {
                System.out.println(prime);
            }
        }
        
    }
}

File: src/task2.kt
fun primesTo(k: Int): Sequence<Int> {
    val numbers = IntArray(k - 1) { it + 2 }

    for ((i, number) in numbers.withIndex()) {
        if (number == 0) continue
        for (j in i + number until numbers.size step number) {
            numbers[j] = 0
        }
    }

    return numbers.asSequence().filter { it != 0 }
}

fun main(args: Array<String>) {
    val test: Sequence<Int>

    primesTo(100_000).forEach { println(it) }
}